sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StatePartitionReader.scala

```diff
@@ -178,44 +178,75 @@ class StatePartitionReader(
     }
   }
 
+  private def processListStateEntries(stateVarName: String): Iterator[InternalRow] = {
```

Comment:
nit: better to move these new private methods below the definition of iter. I'm even OK to see this after `close()`. Let's keep the important and caller-facing (public) things to show earlier.

Reply:
Or, why not just move this out to SchemaUtil as well, like unifyMapStateRowPair?

After moving this out, I don't think we need to add individual private methods - they are simple enough to inline.

Reply:
Done